flycheck.el

```diff
@@ -7744,7 +7744,12 @@ Otherwise return the previously used cache directory."
 Look up the directory hierarchy from FILE for a directory
 containing a file that matches REGEXP."
   (locate-dominating-file
-   file
+   ;; We must not pass regular file here because
+   ;; `locate-dominating-file' will just pass it into its second
+   ;; lambda argument . But our lambda expects only directories.
+   (if (file-directory-p file)
+       file
+     (file-name-directory file))
```

Comment:
It's not idempotent - it either strips trailing slash or last path element, `/usr/bin/ => /usr/bin`, `/usr/bin => /usr/`.